modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/layout/admin/util/CPDefinitionSitemapURLProvider.java

```diff
@@ -161,13 +168,35 @@ protected void visitLayout(
 				_portal.getClassNameId(CProduct.class),
 				cpDefinition.getCProductId());
 
+		List<FriendlyURLEntryLocalization> friendlyURLEntryLocalizations =
+			_friendlyURLEntryLocalService.getFriendlyURLEntryLocalizations(
+				friendlyURLEntry.getFriendlyURLEntryId());
+
```

Comment:
I thnk you should move this declaration close to the for loop

Reply:
Made the requested change and updated PR.